Firestore/core/src/local/leveldb_index_manager.cc

```diff
@@ -922,10 +926,42 @@ void LevelDbIndexManager::DeleteIndexEntry(const model::Document& document,
   }
 }
 
-// TODO(OrQuery): Implement sub targets properly.
-const std::vector<Target> LevelDbIndexManager::GetSubTargets(
-    const Target& target) const {
-  return {target};
+std::vector<Target> LevelDbIndexManager::GetSubTargets(const Target& target) {
+  auto it = target_to_DNF_subtargets_.find(target);
+  if (it != target_to_DNF_subtargets_.end()) {
+    return it->second;
+  }
+
+  std::vector<Target> subtargets;
+  if (target.filters().empty()) {
+    subtargets.push_back(target);
+  } else {
+    // There is an implicit AND operation between all the filters stored in the
+    // target.
+    std::vector<Filter> filters;
+    for (const auto& filter : target.filters()) {
+      filters.push_back(filter);
+    }
+    std::vector<Filter> dnf = LogicUtils::GetDnfTerms(CompositeFilter::Create(
+        std::move(filters), CompositeFilter::Operator::And));
+
+    for (const Filter& term : dnf) {
+      core::FilterList filter_list;
+      if (term.IsAFieldFilter()) {
+        filter_list = filter_list.push_back(term);
+      } else if (term.IsACompositeFilter()) {
+        for (const auto& filter : (CompositeFilter(term)).filters()) {
+          filter_list = filter_list.push_back(filter);
+        }
+      }
```

Comment:
I cannot use `term` directly. Since for composite filter, the `GetSubTargets` needs the filter list inside composite filter instead of composite filter itself.

Reply:
Android is doing `term.getFilters()`.  😢 this is because we we don't have the `term.get_filters()`? It sounds like we should make one? (again, not in this PR though).